### ISSUE TITLE
[Enhancement]: Add support for Multi Repos

### ISSUE THREAD

**Issue by MaxMoldmann** [Vinzent03/obsidian-git#773]
Adding as an enhancement/feature request.
Benefit:
This would allow sharing of topical notes with more than just one group of people.

This has been requested a long time ago in a question that had been closed: 
https://github.com/Vinzent03/obsidian-git/issues/15

**Comment by Vinzent03** [comment]
Please elaborate what you understand under that feature. There's also #703. May git submodules be already what you need? These are already supported in a way.

**Comment by jpherrerap** [comment]
I would also appreciate this feature since I'd also like to share some notes from my vault via git. Ideally keeping everything in the same vault but different folders. What I see is something like this:

```
My Vault
  |- Personal Notes
    |- .git
    |- Note1.md
  |- Work Notes
    |- .git
    |- Note2.md
  |- Other Note.md
```

with the option of switching which repo to sync on the right hand side (or perhaps detecting it from the currently open note).

**Comment by MaxMoldmann** [comment]
> Please elaborate what you understand under that feature. There's also #703. May git submodules be already what you need? These are already supported in a way.

This and #703 are the same duplicate request. 
The requester of #703 is currently solving the problem by using two obsidian-git plugin copies in his fault to have the ability to have two folders in his fault that have separate git repos.

It would be great if this plugin would be able to support multiple git repos in different folders in one vault instead of just one git repo. I don't think git submodules is the right way to address this. 

Let me know if I should elaborate more!

**Comment by ReessKennedy** [comment]
+1

**Comment by fedemengo** [comment]
> I don't think git submodules is the right way to address this.

@MaxMoldmann why do you think that? I'm currently using submodules for my vaults.
Getting updates/syncs works fine. The only thing that's missing for my use case would be a recursive push/update of nested submodules. Although I understand that requires more manual machinery compared to fetching/pulling.

**Comment by MaxMoldmann** [comment]
My desire is to find a way to collaborate with more than just one  team or set of people. Therefore simply being able to use multiple different folders with git repos would solve this scenario.

**Comment by ReessKennedy** [comment]
> My desire is to find a way to collaborate with more than just one team or set of people. Therefore simply being able to use multiple different folders with git repos would solve this scenario. This is also less complicated to set up.

Yeah, VSCode actually beats Obsidian in this functionality -- one can just pull in multiple repos and the built in GIT functionality tracks all changes and allows for commits across all of them from one UI. This would be neat for the Obsid git plugin.

**Comment by ujikol** [comment]
+1

I tried to work around with submodules. But this cause a lot of painful issues.

**Comment by fedemengo** [comment]
> +1 I also need this function
> 
> I have not used Git Submodules in Obsidian yet. Based on my simple understanding of Git Submodules. If you simply use Git Submodules to connect two unrelated Git repositories. .git directories can be independent of each other. But when you push git in the main repository, the Submodule repository will not be pushed to the remote. You still need to open other programs to push the git repository in Git Submodules. Opening other programs sometimes makes me feel very annoying.
> 
> If the developer does not have this plan, I understand. It may affect the complexity of the project, the difficulty of getting started for new users, and also affect other normal users.

I personally integrate the idea of submodules as sub-vault, in the sense that most of the time I'm working of specific vault contexts that are correcly synched as submodules. Once in a while I work on my main vault, which I still open through Obsidian and have obsidian-git synch up the updates of all nested submodules. 
Sharing this in case it might help someone, but I agree that having this managed automatically by the extension would be ideal.

**Comment by Adrian-at-CrimsonAzure** [comment]
I would also be very interested in this feature, and it's really the only thing that is missing in this wonderful plugin.

For reference, here are my manual git scripts that I use on my phone with Termux + Tasker.
`pull.sh`:
```bash
#!/bin/bash
cd ~/storage/shared/Documents/journal
if ! test -f ../journal.lock ; then
    touch ../journal.lock
    git pull --recurse-submodules
    git submodule update --recursive --remote --rebase
    rm ../journal.lock
else
    echo "journal.lock exists, skipping"
fi
exit 0
```
`backup.sh`:
```bash
#!/bin/bash
cd ~/storage/shared/Documents/journal
if ! test -f ../journal.lock ; then
    touch ../journal.lock
    msg="android vault backup: $(date +'%Y-%m-%d %H-%M-%S')"
    git submodule foreach "git add -A"
    git submodule foreach "git commit -a -m \"$msg\""
    git submodule foreach "git push origin HEAD:main"
    git add -A
    git commit -a -m "$msg"
    git push
    sleep 1
    rm ../journal.lock
else
    echo "journal.lock exists, skipping"
fi
exit 0
```

`git submodule foreach "git push origin HEAD:{submodule branch}"` seems to be what is missing for this plugin.

**Comment by LeviBhme** [comment]
+1 would need this. Our team works on multiple projects with other teams (using separate repositories per project) and would benefit greatly from this.

**Comment by crispyduck00** [comment]
Any update on this? Multiple repos would be great.
